docs/markdown/Syntax.md

```diff
@@ -220,6 +220,26 @@ is_x86 = target.startswith('x86') # boolean value 'true'
 is_bsd = target.to_lower().endswith('bsd') # boolean value 'true'
 ```
 
+#### .substring()
+
+Since 0.56.0, you can extract a substring from a string.
+
+```meson
+# Similar to the Python str[start:end] syntax
+target = 'x86_FreeBSD'
+platform = target.substring(0, 3) # prefix string value 'x86'
+system = target.substring(4) # suffix string value 'FreeBSD'
+```
+
+The method accepts negative values where a negative `start` always returns
+empty string and a negative `end` returns from `start` to `len(string) - end`.
```

Comment:
Negative start also means relative to end, in python `'abc'[-2:-1]` gives `b`.

Reply:
I did not find any clear documentation so I was not sure about this behavior, thanks for the clarification. I hope the doc is now clear enough